### PR TITLE
Get version from build VERSION variable

### PR DESCRIPTION
**Pull request by novocaine**
For the web platform, rather than fetching the version of the app from a file on the server which it was downloaded from, fetch it from process.env.VERSION which is set at compile time.

This ensures that the version actually reflects the bundle which is currently running; this isn't necessarily true otherwise, consider after an update when the static files have been updated but the browser is still running an old version.

It also means we don't have to wait for a network request just to get the app's version.

The file this was previously fetched from is written during packaging at https://github.com/vector-im/element-web/blob/develop/scripts/package.sh#L25 - it's the same `VERSION` build variable but stripped of leading `v` if its a semver, which we replicate at runtime

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

### CODE REVIEW

File: src/vector/platform/WebPlatform.ts [hunk 1]
```diff
@@ -147,12 +150,10 @@ export default class WebPlatform extends VectorBasePlatform {
     }
 
     pollForUpdate = () => {
-        return this.getVersion().then((ver) => {
-            if (this.runningVersion === null) {
-                this.runningVersion = ver;
-            } else if (this.runningVersion !== ver) {
-                if (this.shouldShowUpdate(ver)) {
-                    showUpdateToast(this.runningVersion, ver);
+        return this.getMostRecentVersion() .then((mostRecentVersion) => {
```
Comment (turt2live) [inline src/vector/platform/WebPlatform.ts:153]:
```suggestion
        return this.getMostRecentVersion().then((mostRecentVersion) => {
```

**Review by turt2live — COMMENTED** [review]
where does `process` get mocked for safety in the app? Considering this is running in a browser, we don't typically have access to `process` and I'm a bit concerned about setting the precedent that it'll be available.

**Comment by novocaine** [comment]
> where does `process` get mocked for safety in the app? Considering this is running in a browser, we don't typically have access to `process` and I'm a bit concerned about setting the precedent that it'll be available.

This is created by the [EnvironmentPlugin](https://webpack.js.org/plugins/environment-plugin/) previously added for sentry support [here](https://github.com/vector-im/element-web/blob/develop/webpack.config.js#L548). We already refer to this at runtime [here](https://github.com/matrix-org/matrix-react-sdk/blob/3defb863b35b707cfe719b334030a6c66a252261/src/sentry.ts#L224).

**Review by turt2live — APPROVED** [review]
Slightly scary that we use `process` in web code, but I suppose it hasn't bitten us yet.

**Comment by novocaine** [comment]
> Slightly scary that we use `process` in web code, but I suppose it hasn't bitten us yet.

I agree its weird, but this does seem to be the idiom for piping things like this via webpack. I guess the alternative would be to write a .ts file that just contains the version declaration and can be imported

**Pull request merged**